### PR TITLE
feat: handle masked api keys

### PR DESCRIPTION
**Pull request by JeevaRamu0104**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Allow connector cred to update in the preview page itself
https://github.com/user-attachments/assets/0f747203-afb8-405e-bc9d-ae742556df5e

will address FRM separately

## How did you test it?
Configure a connector.
Update the connector.

Test the prod connectors
STRIPE
ADYEN
AUTHORIZEDOTNET
PAYPAL
BANKOFAMERICA
BLUESNAP
BAMBORA
BRAINTREE
CHECKOUT
CRYPTOPAY
CASHTOCODE
CYBERSOURCE
IATAPAY
KLARNA
MIFINITY
NMI
PAYME
TRUSTPAY
VOLT
ZSL
ZEN

Threeds 
NETCETERA

## Where to test it?

- [x] INTEG
- [ ] SANDBOX
- [ ] PROD

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/ThreeDsProcessors/ThreeDsProcessorHome.res [hunk 1]
```diff
@@ -272,6 +272,8 @@ let make = () => {
               connector=connectorName
               setScreenState={_ => ()}
               isPayoutFlow=false
+              setCurrentStep={_ => ()}
```
Comment (PritishBudhiraja) [inline src/screens/ThreeDsProcessors/ThreeDsProcessorHome.res:275]: can we make this optional parameter as it won't be much work

File: src/screens/ThreeDsProcessors/ThreeDsProcessorHome.res [hunk 2]
```diff
@@ -272,6 +272,8 @@ let make = () => {
...
+              updateStepValue=None
```
Comment (PritishBudhiraja) [inline src/screens/ThreeDsProcessors/ThreeDsProcessorHome.res:276]: This value by default something like None

File: src/screens/Connectors/ConnectorUpdateAuthCreds.res [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    <Modal
+      closeOnOutsideClick=true
+      modalHeading={`Update Connector ${connector}`}
+      showModal
+      setShowModal=setShowFeedbackModal
+      childClass="p-1"
+      borderBottom=true
+      revealFrom=Reveal.Right
+      modalClass="w-full md:w-1/3 !h-full overflow-y-scroll !overflow-x-hidden rounded-none text-jp-gray-900">
+      {<>
```
Comment (PritishBudhiraja) [inline src/screens/Connectors/ConnectorUpdateAuthCreds.res:122]: why <>?

**Review by sagarnaikjuspay — DISMISSED** [review]

File: src/screens/Connectors/ConnectorUpdateAuthCreds.res [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  ) = getConnectorFields(connectorDetails)
+
+  let initialValues = React.useMemo(() => {
+    [
+      ("connector_type", connectorInfo.connector_type->JSON.Encode.string),
+      (
+        "connector_account_details",
+        [("auth_type", connectorInfo.connector_account_details.auth_type->JSON.Encode.string)]
+        ->Dict.fromArray
+        ->JSON.Encode.object,
```
Comment (Riddhiagrawal001) [inline src/screens/Connectors/ConnectorUpdateAuthCreds.res:57]: LogicUtils.getJsonFromArrayOfJson

File: src/screens/Connectors/ConnectorUpdateAuthCreds.res [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      ),
+      ("connector_webhook_details", connectorInfo.connector_webhook_details),
+      ("connector_label", connectorInfo.connector_label->JSON.Encode.string),
+      ("metadata", connectorInfo.metadata),
+      ("additional_merchant_data", connectorInfo.additional_merchant_data),
+    ]
+    ->Dict.fromArray
+    ->JSON.Encode.object
```
Comment (Riddhiagrawal001) [inline src/screens/Connectors/ConnectorUpdateAuthCreds.res:65]: LogicUtils.getJsonFromArrayOfJson

File: src/screens/Connectors/ConnectorUpdateAuthCreds.res [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+          connectorMetaDataFields
+          connectorWebHookDetails
+          connectorLabelDetailField
+          connectorAdditionalMerchantData
+        />
+        <div className="flex p-1 justify-end mb-2">
+          <FormRenderer.SubmitButton text="Submit" />
+        </div>
+        <FormValuesSpy />
+      </Form>}
```
Comment (Riddhiagrawal001) [inline src/screens/Connectors/ConnectorUpdateAuthCreds.res:146]: Why do we need to add {}

File: src/utils/Mappers/ConnectorListMapper.res [hunk 7]
```diff
@@ -125,6 +125,15 @@ let getProcessorPayloadType = dict => {
     merchant_connector_id: dict->getString("merchant_connector_id", ""),
     frm_configs: dict->getArrayFromDict("frm_configs", [])->convertFRMConfigJsonToObjResponse,
     status: dict->getString("status", "inactive"),
+    connector_webhook_details: dict
+    ->Dict.get("connector_webhook_details")
+    ->Option.getOr(JSON.Encode.null),
+    metadata: dict
+    ->Dict.get("metadata")
+    ->Option.getOr(JSON.Encode.null),
```
Comment (Riddhiagrawal001) [inline src/utils/Mappers/ConnectorListMapper.res:133]: LogicUtils.getObj("metadata",JSON.Encode.null)

File: src/screens/ThreeDsProcessors/ThreeDsProcessorHome.res (hunk 1 above)
Comment (JeevaRamu0104) [reply src/screens/ThreeDsProcessors/ThreeDsProcessorHome.res:275]: we will  remove the whole quick start module.

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Pull request merged**